### PR TITLE
KHKWE-33-implement-firebase-login

### PR DESCRIPTION
**Pull request by gamesminer**
screencapture-localhost-8080-login-2018-11-06-13_11_53

Only layout. It isn`t connected AuthentificationService.

### CODE REVIEW

**Comment by ihor4568** [comment]
Please shorten `Authentification` in all names to just `Auth`

File: src/components/Main/Main.html [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <div class="main">
   <aside class="main__sidebar mdc-drawer mdc-drawer--dismissible">
     <div class="mdc-drawer__header">
-      <h3 class="mdc-drawer__title">Library</h3>
-      <h6 class="mdc-drawer__subtitle">Discover & Listen</h6>
+      <h3 class="mdc-drawer__title main__user">
+        <%= currentUser %>
+      </h3>
+      <a href="login" class="mdc-drawer__subtitle main__sign-out">Sign Out</a>
```
Comment (ihor4568) [inline src/components/Main/Main.html]: '/login'

File: src/components/Main/Main.js [hunk 2]
```diff
@@ -81,7 +102,22 @@ export class MainComponent {
...
 
-    if (pathname === "") {
+    const urlParts = pathname.split("/");
+    if (urlParts[0] === "song" && urlParts[1] && urlParts.length === 2) {
+      const songId = urlParts[1];
+      this.setShareView(songId);
+      return;
+    }
+    AuthService.check().then(
+      user => this.handleGo.call(this, pathname, user),
```
Comment (ihor4568) [inline src/components/Main/Main.js]: please read docs for bind and call usage -- and fix these

File: src/components/Auth/Auth.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    this.next.addEventListener("click", this.handleNextClick.bind(this));
+  }
+
+  handleSignInTabClick() {
+    if (this.tabSignIn.classList.contains("auth__input--active") === false) {
+      this.name.classList.add("auth__elem_disable");
+      this.password[1].classList.add("auth__elem_disable");
+      this.tabSignIn.classList.add("auth__input--active");
+      this.tabSignUp.classList.remove("auth__input--active");
+      this.input[0].value = "";
```
Comment (ihor4568) [inline src/components/Auth/Auth.js]: please extract this clearing of inputs into method

File: src/components/Auth/Auth.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+  querySelectors() {
+    this.email = this.mountPoint.querySelector(".auth__email");
+    this.password = this.mountPoint.querySelectorAll(".auth__password");
+    this.name = this.mountPoint.querySelector(".auth__name");
+    this.next = this.mountPoint.querySelector(".auth__button");
+    this.tabBar = this.mountPoint.querySelector(".auth__tab-bar");
+    this.tabSignIn = this.mountPoint.querySelector(".auth__tab-sign-in");
+    this.tabSignUp = this.mountPoint.querySelector(".auth__tab-sign-up");
+    this.input = this.mountPoint.querySelectorAll(".auth__input");
```
Comment (ihor4568) [inline src/components/Auth/Auth.js]: inputs - use plural if it contains multiple elements

File: src/components/Auth/Auth.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+export class AuthComponent {
+  constructor(mountPoint, props = {}) {
+    this.mountPoint = mountPoint;
+    this.props = props;
+  }
+
+  querySelectors() {
+    this.email = this.mountPoint.querySelector(".auth__email");
+    this.password = this.mountPoint.querySelectorAll(".auth__password");
```
Comment (ihor4568) [inline src/components/Auth/Auth.js]: passwords - use plural if it contains multiple elements

File: src/components/Auth/Auth.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    );
+    this.tabSignUp.addEventListener(
+      "click",
+      this.handleSignUpTabClick.bind(this)
+    );
+    this.next.addEventListener("click", this.handleNextClick.bind(this));
+  }
+
+  handleSignInTabClick() {
+    if (this.tabSignIn.classList.contains("auth__input--active") === false) {
```
Comment (ihor4568) [inline src/components/Auth/Auth.js]: if (!this.tabSignIn.classList.contains("auth__input--active"))

File: src/components/Auth/Auth.js [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+      this.input[1].value = "";
+      this.input[2].value = "";
+      this.input[3].value = "";
+      this.erorPoint.innerText = "";
+      this.next.innerText = "SIGN IN";
+    }
+  }
+
+  handleSignUpTabClick() {
+    if (this.tabSignUp.classList.contains("auth__input--active") === false) {
```
Comment (ihor4568) [inline src/components/Auth/Auth.js]: if (!this.tabSignIn.classList.contains("auth__input--active"))

File: src/components/Auth/Auth.js [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        this.handleAlert.call(this, err)
+      );
+    }
+    if (this.next.innerText === "SIGN UP") {
+      if (this.input[2].value === this.input[3].value) {
+        AuthService.signUp(
+          this.input[0].value,
+          this.input[2].value,
+          this.input[1].value,
+          err => this.handleAlert.call(this, err)
```
Comment (ihor4568) [inline src/components/Auth/Auth.js]: please read doc for call and bind and fix this

File: src/components/Auth/Auth.js [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  handleNextClick() {
+    if (this.next.innerText === "SIGN IN") {
+      AuthService.signIn(this.input[0].value, this.input[2].value, err =>
+        this.handleAlert.call(this, err)
+      );
+    }
+    if (this.next.innerText === "SIGN UP") {
+      if (this.input[2].value === this.input[3].value) {
```
Comment (ihor4568) [inline src/components/Auth/Auth.js]: it looks very unreadable -- you'd better use different names for inputs

File: src/components/Auth/Auth.js [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+        AuthService.signUp(
+          this.input[0].value,
+          this.input[2].value,
+          this.input[1].value,
+          err => this.handleAlert.call(this, err)
+        );
+      } else {
+        this.erorPoint.innerText = "auth/passwords must be identical";
```
Comment (ihor4568) [inline src/components/Auth/Auth.js]: please come up with better error message. Also it should start from capital letter

File: src/components/Auth/Auth.js [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+      }
+    }
+  }
+
+  handleAlert(err) {
+    this.erorPoint.innerText = err;
```
Comment (ihor4568) [inline src/components/Auth/Auth.js]: error is written with 2 rr

File: src/services/AuthService.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
+import { FirebaseService } from "./FirebaseService";
+
+const auth = FirebaseService.auth();
+
+export class AuthService {
+  static signIn(email, pass, cb = null) {
```
Comment (ihor4568) [inline src/services/AuthService.js]: you should return promise not callback. Please fix this in all cases

File: src/components/Auth/Auth.js (hunk 6 above)
Comment (gamesminer) [reply src/components/Auth/Auth.js]: Maksym told us that we shoudn't use ! Because it is hard to read

**Review by WOLFRIEND — APPROVED** [review]

**Pull request merged**